src/Guides/RestructuredText/Parser/DocumentParser.php

```diff
@@ -505,15 +505,9 @@ private function flush(): void
 
             if ($currentDirective !== null) {
                 try {
-                    // if a Directive is not followed by a CodeNode; this means it totally relies on
-                    // its data and we should not process the $node as part of handling this directive
-                    $directiveNode = $node instanceof CodeNode
-                            ? $node
-                            : new SpanNode($this->environment, $this->directive->getData());
```

Comment:
this has been removed? I remember you introduced this recently? when introducing the inline uml?

Reply:
> Additionally, I found a slight bug in the DocumentParser; when a
Directive is followed by something other than a CodeNode we want to
ignore it and not pass it as a SpanNode to the directive's process
function. In that scenario, the deeper business logic will convert that
SpanNode into a piece of text on the page. Which you do not want.

Reply:
Yeah, I found out it causes a weird situation with single-line directives